unable to connect to cluster if first node is down

# problem description
I have created a quarkus app connecting to a hornetq cluster, and sending a message every 5 seconds.
if both hornetq nodes are up, and I start the application, I can send my messages OK.
if I shut down node `x1`, and I start the application, no messages get sent at all, and the following exception gets logged every time the scheduler gets called, and no messages get sent:

```
2024-07-23 15:50:37,869 INFO  [io.qua.iro.runtime] (jca-worker-pool-<default>-1) QIJ000001: Starting Resource Adapter <default>: ActiveMQ Artemis 2.33.0

2024-07-23 15:50:37,884 INFO  [org.apa.act.art.ra.ActiveMQRALogger] (jca-worker-pool-<default>-1) AMQ151007: Resource adaptor started
2024-07-23 15:50:38,041 INFO  [io.quarkus] (Quarkus Main Thread) jmsraavail 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.12.3) started in 3.446s. Listening on: http://localhost:18080                                              
2024-07-23 15:50:38,042 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2024-07-23 15:50:38,043 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [artemis-core, artemis-jms-ra, cdi, ironjacamar, narayana-jta, quartz, rest, scheduler, smallrye-context-propagation, vertx]    
...              
2024-07-23 15:50:40,145 WARN  [org.jbo.jca.cor.con.poo.str.PoolByCri] (vert.x-worker-thread-1) IJ000604: Throwable while attempting to get a new connection: null: jakarta.resource.ResourceException: Failed to create session factory
        at org.apache.activemq.artemis.ra.ActiveMQRAManagedConnection.setup(ActiveMQRAManagedConnection.java:725)
        at org.apache.activemq.artemis.ra.ActiveMQRAManagedConnection.<init>(ActiveMQRAManagedConnection.java:161)
        at org.apache.activemq.artemis.ra.ActiveMQRAManagedConnectionFactory.createManagedConnection(ActiveMQRAManagedConnectionFactory.java:144)
        at org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreArrayListManagedConnectionPool.createConnectionEventListener(SemaphoreArrayListManagedConnectionPool.java:1267)
        at org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreArrayListManagedConnectionPool.getConnection(SemaphoreArrayListManagedConnectionPool.java:495)
        at org.jboss.jca.core.connectionmanager.pool.AbstractPool.getTransactionNewConnection(AbstractPool.java:770)
        at org.jboss.jca.core.connectionmanager.pool.AbstractPool.getConnection(AbstractPool.java:666)
        at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.getManagedConnection(AbstractConnectionManager.java:624)
        at org.jboss.jca.core.connectionmanager.tx.TxConnectionManagerImpl.getManagedConnection(TxConnectionManagerImpl.java:440)
        at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.allocateConnection(AbstractConnectionManager.java:789)
        at org.apache.activemq.artemis.ra.ActiveMQRASessionFactoryImpl.allocateConnection(ActiveMQRASessionFactoryImpl.java:792)
        at org.apache.activemq.artemis.ra.ActiveMQRASessionFactoryImpl.createSession(ActiveMQRASessionFactoryImpl.java:482)
        at org.apache.activemq.artemis.ra.ActiveMQRASessionFactoryImpl.createSession(ActiveMQRASessionFactoryImpl.java:673)
        at org.apache.activemq.artemis.ra.ActiveMQRASessionFactoryImpl.createSession(ActiveMQRASessionFactoryImpl.java:678)
        at org.apache.activemq.artemis.ra.ActiveMQRAConnectionFactoryImpl.validateUser(ActiveMQRAConnectionFactoryImpl.java:422)
        at org.apache.activemq.artemis.ra.ActiveMQRAConnectionFactoryImpl.createContext(ActiveMQRAConnectionFactoryImpl.java:379)
        at org.apache.activemq.artemis.ra.ActiveMQRAConnectionFactoryImpl.createContext(ActiveMQRAConnectionFactoryImpl.java:369)
        at org.apache.activemq.artemis.ra.ActiveMQRAConnectionFactoryImpl.createContext(ActiveMQRAConnectionFactoryImpl.java:364)
        at org.acme.MyProducer.send(MyProducer.java:40)
        at org.acme.MyProducer_Subclass.send$$superforward(Unknown Source)
        at org.acme.MyProducer_Subclass$$function$$1.apply(Unknown Source)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:136)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:107)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:38)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:61)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:32)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
        at org.acme.MyProducer_Subclass.send(Unknown Source)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createConnectionInternal(ActiveMQConnectionFactory.java:915)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createXAConnection(ActiveMQConnectionFactory.java:387)
        at org.apache.activemq.artemis.ra.ActiveMQRAManagedConnection.setup(ActiveMQRAManagedConnection.java:713)
        ... 49 more
Caused by: ActiveMQNotConnectedException[errorType=NOT_CONNECTED message=AMQ219007: Cannot connect to server(s). Tried with all available servers.]
        at org.apache.activemq.artemis.core.client.impl.ServerLocatorImpl.createSessionFactory(ServerLocatorImpl.java:735)
        at org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.createConnectionInternal(ActiveMQConnectionFactory.java:913)
        ... 51 more
```

The application has the following configuration:
```
quarkus.http.port=18080
quarkus.ironjacamar.ra.kind=artemis
quarkus.ironjacamar.ra.config.connection-parameters=host=x1;port=...,host=x2;port=...;protocols=CORE
quarkus.ironjacamar.ra.config.protocol-manager-factory=org.apache.activemq.artemis.core.protocol.hornetq.client.HornetQClientProtocolManagerFactory
quarkus.ironjacamar.ra.config.user=...
quarkus.ironjacamar.ra.config.password=...
quarkus.ironjacamar.activation-spec.myqueue.config.destination-type=jakarta.jms.Queue
quarkus.ironjacamar.activation-spec.myqueue.config.destination=jms.queue....
quarkus.ironjacamar.activation-spec.myqueue.config.max-session=10
quarkus.ironjacamar.activation-spec.myqueue.config.rebalance-connections=true
```

if I exchange `x1` and `x2` then it works:
```
quarkus.ironjacamar.ra.config.connection-parameters=host=x2;port=...,host=x1;port=...;protocols=CORE
```

this only fails when the failed node is first in the url.

# expectation

the client should be able to connect to any running node.

# producer code
```
@ApplicationScoped
public class MyProducer {

    private static final Logger log = LoggerFactory.getLogger(MyProducer.class);

    public static final String QUEUE = "jms.queue.com.lodh.jee.ArteTest.Bank.jms.bankref1.QueueB";

    private final Queue queue = ActiveMQDestination.createQueue(QUEUE);;

    @Inject
    ConnectionFactory factory;

    @Scheduled(every = "5s")
    public void send() throws UnknownHostException {
        try (JMSContext context = factory.createContext()) {
            String text = new Date() + " from host=" + InetAddress.getLocalHost().getHostName();
            log.info("sending jms message {}", text);
            JMSProducer producer = context.createProducer();
            TextMessage message = context.createTextMessage(text);
            producer.send(queue, message);
        }
    }
}
```

# pom
```
<quarkus.platform.version>3.12.3</quarkus.platform.version>
...
    <dependency>
      <groupId>io.quarkiverse.artemis</groupId>
      <artifactId>quarkus-artemis-jms-ra</artifactId>
      <version>3.3.0</version>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-scheduler</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.activemq</groupId>
      <artifactId>artemis-hqclient-protocol</artifactId>
      <version>2.33.0</version>
    </dependency>
```

cc @gastaldi @zhfeng

--- a/ra/runtime/src/main/java/io/quarkus/artemis/jms/ra/ArtemisResourceAdapterFactory.java
+++ b/ra/runtime/src/main/java/io/quarkus/artemis/jms/ra/ArtemisResourceAdapterFactory.java
@@ -1,6 +1,8 @@
 package io.quarkus.artemis.jms.ra;
 
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 import jakarta.jms.ConnectionFactory;
 import jakarta.jms.Message;
@@ -43,8 +45,14 @@ public class ArtemisResourceAdapterFactory implements ResourceAdapterFactory {
     @Override
     public ActiveMQResourceAdapter createResourceAdapter(String id, Map<String, String> config) {
         ActiveMQResourceAdapter adapter = new ActiveMQResourceAdapter();
-        adapter.setConnectorClassName(NettyConnectorFactory.class.getName());
-        adapter.setConnectionParameters(config.get("connection-parameters"));
+        String connectionParameters = config.get("connection-parameters");
+        int hosts = Math.max(1, count(connectionParameters, "host="));
+        // Repeat the NettyConnectorFactory class name for each host
+        String connectorClassName = NettyConnectorFactory.class.getName();
+        adapter.setConnectorClassName(IntStream.rangeClosed(1, hosts)
+                .mapToObj(unused -> connectorClassName)
+                .collect(Collectors.joining(",")));
+        adapter.setConnectionParameters(connectionParameters);
         adapter.setProtocolManagerFactoryStr(config.get("protocol-manager-factory"));
         adapter.setUseJNDI(false);
         adapter.setUserName(config.get("user"));
@@ -84,6 +92,15 @@ public class ArtemisResourceAdapterFactory implements ResourceAdapterFactory {
     @Override
     public MessageEndpoint wrap(MessageEndpoint endpoint, Object resourceEndpoint) {
         return new JMSMessageEndpoint(endpoint, (MessageListener) resourceEndpoint);
+    }
+
+    private static int count(String text, String find) {
+        int count = 0;
+        final int length = find.length();
+        for (int index = 0; (index = text.indexOf(find, index)) != -1; index += length) {
+            count++;
+        }
+        return count;
     }
 
     private static class JMSMessageEndpoint extends MessageEndpointWrapper implements MessageListener {

